include/aws/io/event_loop.h

```diff
@@ -6,286 +6,50 @@
- * While the I/O is pending, it is not safe to modify or delete aws_overlapped.
- * Call aws_overlapped_init() before first use. If the aws_overlapped will be used multiple times, call
- * aws_overlapped_reset() or aws_overlapped_init() between uses.
+ * Event loop group configuration options
  */
-struct aws_overlapped {
-    struct aws_win32_OVERLAPPED overlapped;
-    aws_event_loop_on_completion_fn *on_completion;
-    void *user_data;
-};
-
-#else /* !AWS_USE_IO_COMPLETION_PORTS */
-
-typedef void(aws_event_loop_on_event_fn)(
-    struct aws_event_loop *event_loop,
-    struct aws_io_handle *handle,
-    int events,
-    void *user_data);
-
-#endif /* AWS_USE_IO_COMPLETION_PORTS */
-
-struct aws_event_loop_vtable {
-    void (*destroy)(struct aws_event_loop *event_loop);
-    int (*run)(struct aws_event_loop *event_loop);
-    int (*stop)(struct aws_event_loop *event_loop);
-    int (*wait_for_stop_completion)(struct aws_event_loop *event_loop);
-    void (*schedule_task_now)(struct aws_event_loop *event_loop, struct aws_task *task);
-    void (*schedule_task_future)(struct aws_event_loop *event_loop, struct aws_task *task, uint64_t run_at_nanos);
-    void (*cancel_task)(struct aws_event_loop *event_loop, struct aws_task *task);
-#if AWS_USE_IO_COMPLETION_PORTS
-    int (*connect_to_io_completion_port)(struct aws_event_loop *event_loop, struct aws_io_handle *handle);
-#else
-    int (*subscribe_to_io_events)(
-        struct aws_event_loop *event_loop,
-        struct aws_io_handle *handle,
-        int events,
-        aws_event_loop_on_event_fn *on_event,
-        void *user_data);
-#endif
-    int (*unsubscribe_from_io_events)(struct aws_event_loop *event_loop, struct aws_io_handle *handle);
-    void (*free_io_event_resources)(void *user_data);
-    bool (*is_on_callers_thread)(struct aws_event_loop *event_loop);
-};
-
-struct aws_event_loop {
-    struct aws_event_loop_vtable *vtable;
-    struct aws_allocator *alloc;
-    aws_io_clock_fn *clock;
-    struct aws_hash_table local_data;
-    struct aws_atomic_var current_load_factor;
-    uint64_t latest_tick_start;
-    size_t current_tick_latency_sum;
-    struct aws_atomic_var next_flush_time;
-    void *impl_data;
-};
+struct aws_event_loop_group_options {
 
-struct aws_event_loop_local_object;
-typedef void(aws_event_loop_on_local_object_removed_fn)(struct aws_event_loop_local_object *);
+    /**
+     * How many event loops that event loop group should contain.  For most group types, this implies
+     * the creation and management of an analagous amount of managed threads
+     */
+    uint16_t loop_count;
 
-struct aws_event_loop_local_object {
-    const void *key;
-    void *object;
-    aws_event_loop_on_local_object_removed_fn *on_object_removed;
-};
+    /**
+     * Optional callback to invoke when the event loop group finishes destruction.
+     */
+    struct aws_shutdown_callback_options *shutdown_options;
```

Comment:
utterly trivial: const pointers
```suggestion
    const struct aws_shutdown_callback_options *shutdown_options;
```